archivist_samples/c2pa/c2pa.py

```diff
@@ -0,0 +1,320 @@
+            "arc_blob_identity": blob["identity"],
+            "arc_blob_hash_alg": blob["hash"]["alg"],
+            "arc_blob_hash_value": blob["hash"]["value"],
+        }
+        return attachment
+
+
+def attachment_create(arch, attachment_description: AttachmentDescription):
+    with resources.open_binary(c2pa_files, attachment_description.filename) as fd:
+        attachment = arch.attachments.upload(fd)
+        result = {
+            "arc_attribute_type": "arc_attachment",
+            "arc_blob_identity": attachment["identity"],
+            "arc_blob_hash_alg": attachment["hash"]["alg"],
+            "arc_blob_hash_value": attachment["hash"]["value"],
+            "arc_display_name": attachment_description.attribute_name,
+            "arc_file_name": attachment_description.filename,
+        }
+
+        return result
+
+
+document_creator = make_assets_create(
+    attachment_creator=attachment_create, confirm=True
+)
+
+
+class C2PADocument:
+    selector_key = "OnboardingSampleID"
+    selector_value = "C2PA"
+
+    def __init__(
+        self,
+        arch: "Archivist",
+        display_type: str,
+    ):
+        arch_ = copy(arch)
+        arch_.fixtures = {
+            "assets": {
+                "attributes": {
+                    "arc_display_type": display_type,
+                },
+            },
+        }
+
+        self._arch = arch_
+        self._asset = None
+        self._existed = False
+
+    @property
+    def arch(self):
+        return self._arch
+
+    @property
+    def asset(self):
+        return self._asset
+
+    @property
+    def existed(self):
+        return self._existed
+
+    def create(
+        self,
+        name: str,
+        description: str,
+        identity: str,
+        *,
+        attachments: "list|None" = None,
+        custom_attrs: "dict|None" = None,
+    ):
+        attrs = {
+            "arc_description": description,
+            "arc_profile": "Document",
+            "id": identity,
+        }
+
+        if custom_attrs is not None:
+            attrs.update(custom_attrs)
+
+        self._asset, self._existed = document_creator(
+            self.arch,
+            name,
+            attrs,
+            attachments=attachments,
+            selector_key=self.selector_key,
+            selector_value=self.selector_value,
+        )
+
+        return self._asset
+
+    # Publish new version of the document
+    # pylint: disable=too-many-arguments
+    def publish(
+        self,
+        document: dict,
+        version: str,
+        description: str,
+        doc_hash: str,
+        authors: "list[dict]",
+        name: str,
+        custom_attrs: "dict|None" = None,
+    ):
+        props = {
+            "operation": "Record",
+            "behaviour": "RecordEvidence",
+        }
+        attrs = {
+            "arc_display_type": "Publish",
+            "arc_description": description,
+            "document_version_authors": authors,
```

Comment:
Something to note here is that we have weak validation for this in the backend. There is a feature about tightening that up - but users could enter some dodgy values for this. I believe the documentation on the website tells them how it should be created though.

Reply:
I think that is a superb idea :) It took me by surprise the format of authors. I will document the format.

Reply:
have added the expected authors format to the function comment.